constraint/pkg/client/targethandler.go

```diff
@@ -0,0 +1,99 @@
+package client
+
+import (
+	"text/template"
+
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type TargetHandler interface {
+	MatchSchemaProvider
+
+	// GetName returns name of the target. Must match `^[a-zA-Z][a-zA-Z0-9.]*$`
+	// This will be the exact name of the field in the ConstraintTemplate
+	// spec.target object, so if GetName returns validation.xyz.org, the user
+	// will populate target specific rego into .spec.targets."validation.xyz.org".
+	GetName() string
+
+	// Library returns the pieces of Rego code required to stitch together constraint
+	// evaluation for the target.
+	//
+	// Templated Parameters:
+	//	`{{.ConstraintsRoot}}`: All constraints are found here and are placed under
+	//		data["{{.ConstraintsRoot}}"][KIND][NAME] where $KIND is the Kind indicated in the
+	//		CT .spec.crd.spec.names.kind field and NAME is the .metadata.name of the
+	//		Constraint resource.
+	//	`{{.DataRoot}}`: The data root, use data["{{.DataRoot}}"] instead of
```

Comment:
Correct. Side note: the data is actually cached in a target-specific path, so `data.inventory` only exists at template execution time.

(no action needed)